src/Engine/TranskribusClient.php

```diff
@@ -0,0 +1,277 @@
+        $url = self::PROCESSES_URL . '/'. $this->processId;
+        $response = $this->httpClient
+        ->request(
+            'GET',
+            $url,
+            [
+                'headers' => [
+                    'Content-Type' => 'application/json',
+                    'Authorization' => 'Bearer '.$this->accessToken
+                ],
+            ]
+        );
+        
+        $statusCode = $response->getStatusCode();
+       
+        if (200 === $statusCode) {
+            $content = json_decode($response->getContent());
+            if (is_null($content)) {
+                $message = 'Transkribus API returned null';
+                $this->setErrorMessage($message);
+            } else {
+                
+                if($content->{'status'} === 'FINISHED'){
+                    $textContent = $content->{'content'};
+                    $this->textResult = $textContent->{'text'};
+                    $this->hasError = false;
+                }                
+            }
+        } elseif (401 === $statusCode) {
+            $this->refereshAccessToken();
+            if($this->getHasError()){
+                $message = 'Error Code '.$statusCode.' :: Credentials are incorrect!';
+                $this->setErrorMessage($message);
+            }else{
+                $this->retryProcessStatus();
+            }
+
+        } else {
+            $rawResponse   = $response->getContent(false);
+            $message = 'Error Code '.$statusCode.' :: Error retrieving process status, try again!';
+            $this->setErrorMessage($message);
+        }
+    }
+
+    public function retryProcessStatus(){
+        $this->retrieveProcessStatus();
+    }
+
+    public function refereshAccessToken(){
+        if(!file_exists('transkribus_credentials.json')){
+            $this->setAccessToken();
+            
+        }else{
+            $data = file_get_contents('transkribus_credentials.json');
+            $content = json_decode($data);
+            $refreshToken = $content->{'refresh_token'};
+            $response = $this->httpClient
+            ->request(
+                'POST',
+                self::AUTHENTICATION_URL,
+                [
+                    'headers' => [
+                        'Content-Type' => 'application/x-www-form-urlencoded',
+                    ],
+                    'body' => [
+                        'grant_type' => 'refresh_token',
+                        'username' => $this->credentials['username'],
+                        'client_id' => 'processing-api-client',
+                        'refresh_token' => $refreshToken,
+                    ],
+                ]
+            );
+            $statusCode = $response->getStatusCode();
+            if (200 === $statusCode) {
+                $content = json_decode($response->getContent());
+
+                if (is_null($content)) {
+                    $message = 'Transkribus API returned null';
```

Comment:
- Need to modify the error message again
- I think this is missing a call to `setErrorMessage()`